NuRadioMC/EvtGen/NuRadioProposal.py

```diff
@@ -40,6 +40,10 @@
 pp_km = 1.e5
 
 
+def calculate_distance(pp_position, pos_arr):
```

Comment:
indeed makes code much nicer, but unfortunately also WRONG if I am not mistaken! ;) pp_position is in cm!

Reply:
You are right, `pp_position` is in proposal units. But I thought `pos_arr`, which (after checking) seems not to be the case... 
I just tried to replace the old code which is the one below which is identical (wrong?) to what I wrote, right?
```
distance = ((sec.position.x - lepton_position[0]) * units.cm) ** 2
distance += ((sec.position.y - lepton_position[1]) * units.cm) ** 2
distance += ((sec.position.z - lepton_position[2]) * units.cm) ** 2
distance = np.sqrt(distance)
```

Reply:
Apropos units. I do not quite understand where we should use proposal units and where NuRadioMC units. Things like `SecondaryProperties` are in NuRadioMC units. So the distances should be converted to them (i.e. multiplied with units.cm). I think the problem is that the `lepton_position` has not been converted to proposal units?

Reply:
And how could that not be catched by the CI or did it?

Reply:
oh sorry, my bad. yes you were indeed right! Which underlines that your code is much nicer to read now. Thanks!

Reply:
concerning units, it depends if it is an internal function starting with `__` or not:
```
class ProposalFunctions(object):
    """
    This class serves as a container for PROPOSAL functions. The functions that
    start with double underscore take PROPOSAL units as an argument and should
    not be used from the outside to avoid mismatching units.
    """
```

Reply:
But now I am confused: `lepton_positions_nu` is not converted into proposal units (i.e., cm) at the beginning of function `get_secondaries_array` but it should?

Reply:
And should we rename `calculate_distance` to `__calculate_distance` because it takes proposal units? But then it returns NuRadioMC units which make sense when the distances it calculates it returns but its awkward.